### ISSUE TITLE
AttributeError: module 'numpy' has no attribute 'float' when plotting selected FRET pairs

### ISSUE THREAD

**Issue by Frank-Duffy** [SMB-Lab/FRETNet-Designer#4]
I was testing George's installation instructions in the readme in a fresh environment when I encountered this error. It appears that there is a dependency version issue. the matplotlib version 3.2.2 I installed uses np.float, which was deprecated in NumPy 1.20, leading to the error.

Error in the "plot the selected fret pairs on a radial plot" cell below:

---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[111], line 3
      1 #plot the selected FRET pairs on a radial plot
----> 3 ax=plt.subplot(111,projection='polar')
      4 firstres=single_residue_info['Residue Index'].iloc[0]
      5 lastres=single_residue_info['Residue Index'].iloc[-1]

File ~/.conda/envs/new_fret_dev/lib/python3.8/site-packages/matplotlib/pyplot.py:1030, in subplot(*args, **kwargs)
   1025     cbook._warn_external("The subplot index argument to subplot() appears "
   1026                          "to be a boolean. Did you intend to use "
   1027                          "subplots()?")
   1029 fig = gcf()
-> 1030 a = fig.add_subplot(*args, **kwargs)
   1031 bbox = a.bbox
   1032 byebye = []

File ~/.conda/envs/new_fret_dev/lib/python3.8/site-packages/matplotlib/figure.py:1419, in Figure.add_subplot(self, *args, **kwargs)
   1411         else:
   1412             # Undocumented convenience behavior:
   1413             # subplot(111); subplot(111, projection='polar')
   1414             # will replace the first with the second.
   1415             # Without this, add_subplot would be simpler and
   1416             # more similar to add_axes.
   1417             self._axstack.remove(ax)
-> 1419     a = subplot_class_factory(projection_class)(self, *args, **kwargs)
   1421 return self._add_axes_internal(key, a)

File ~/.conda/envs/new_fret_dev/lib/python3.8/site-packages/matplotlib/axes/_subplots.py:76, in SubplotBase.__init__(self, fig, *args, **kwargs)
     73 self.update_params()
     75 # _axes_class is set in the subplot_class_factory
---> 76 self._axes_class.__init__(self, fig, self.figbox, **kwargs)
     77 # add a layout box to this, for both the full axis, and the poss
     78 # of the axis.  We need both because the axes may become smaller
     79 # due to parasitic axes and hence no longer fill the subplotspec.
     80 if self._subplotspec._layoutbox is None:

File ~/.conda/envs/new_fret_dev/lib/python3.8/site-packages/matplotlib/projections/polar.py:795, in PolarAxes.__init__(self, theta_offset, theta_direction, rlabel_position, *args, **kwargs)
    793 self._default_theta_direction = theta_direction
    794 self._default_rlabel_position = np.deg2rad(rlabel_position)
--> 795 super().__init__(*args, **kwargs)
    796 self.use_sticky_edges = True
    797 self.set_aspect('equal', adjustable='box', anchor='C')

File ~/.conda/envs/new_fret_dev/lib/python3.8/site-packages/matplotlib/axes/_base.py:439, in _AxesBase.__init__(self, fig, rect, facecolor, frameon, sharex, sharey, label, xscale, yscale, **kwargs)
    437     self._shared_y_axes.join(self, sharey)
    438 self.set_label(label)
--> 439 self.set_figure(fig)
    441 self.set_axes_locator(kwargs.get("axes_locator", None))
    443 self.spines = self._gen_axes_spines()

File ~/.conda/envs/new_fret_dev/lib/python3.8/site-packages/matplotlib/axes/_base.py:581, in _AxesBase.set_figure(self, fig)
    577 self._viewLim = mtransforms.Bbox.unit()
    578 self.transScale = mtransforms.TransformWrapper(
    579     mtransforms.IdentityTransform())
--> 581 self._set_lim_and_transforms()

File ~/.conda/envs/new_fret_dev/lib/python3.8/site-packages/matplotlib/projections/polar.py:858, in PolarAxes._set_lim_and_transforms(self)
    852 self.transScale = mtransforms.TransformWrapper(
    853     mtransforms.IdentityTransform())
    855 # Scale view limit into a bbox around the selected wedge. This may be
    856 # smaller than the usual unit axes rectangle if not plotting the full
    857 # circle.
--> 858 self.axesLim = _WedgeBbox((0.5, 0.5),
    859                           self._realViewLim, self._originViewLim)
    861 # Scale the wedge to fill the axes.
    862 self.transWedge = mtransforms.BboxTransformFrom(self.axesLim)

File ~/.conda/envs/new_fret_dev/lib/python3.8/site-packages/matplotlib/projections/polar.py:729, in _WedgeBbox.__init__(self, center, viewLim, originLim, **kwargs)
    727 def __init__(self, center, viewLim, originLim, **kwargs):
    728     mtransforms.Bbox.__init__(self,
--> 729                               np.array([[0.0, 0.0], [1.0, 1.0]], np.float),
    730                               **kwargs)
    731     self._center = center
    732     self._viewLim = viewLim

File ~/.conda/envs/new_fret_dev/lib/python3.8/site-packages/numpy/__init__.py:305, in __getattr__(attr)
    300     warnings.warn(
    301         f"In the future `np.{attr}` will be defined as the "
    302         "corresponding NumPy scalar.", FutureWarning, stacklevel=2)
    304 if attr in __former_attrs__:
--> 305     raise AttributeError(__former_attrs__[attr])
    307 # Importing Tester requires importing all of UnitTest which is not a
    308 # cheap import Since it is mainly used in test suits, we lazy import it
    309 # here to save on the order of 10 ms of import time for most users
    310 #
    311 # The previous way Tester was imported also had a side effect of adding
    312 # the full `numpy.testing` namespace
    313 if attr == 'testing':

AttributeError: module 'numpy' has no attribute 'float'.
`np.float` was a deprecated alias for the builtin `float`. To avoid this error in existing code, use `float` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.float64` here.
The aliases was originally deprecated in NumPy 1.20; for more details and guidance see the original release note at:
    https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations

**Comment by Frank-Duffy** [comment]
My plan is to check the versions of the dependencies involved in the readme instructions test environment and compare them to env in which this error did not occur. I believe simply updating matplotlib will fix this issue, but I'm not sure yet what version we need. Obtaining the last working version may give some insight.

These are the dependency versions that resulted from following the readme on a fresh install.

fretraj: 0.2.10
numpy: 1.24.4
numba: 0.58.1
matplotlib: 3.2.2
python: 3.8.0 | packaged by conda-forge | (default, Nov 22 2019, 19:11:38) 
[GCC 7.3.0]

**Comment by Frank-Duffy** [comment]
Unfortunately it seems that my original development environment is no longer working after creating this new one... I will update matplotlib to the latest version that supports python 3.8 in the new environment and see if that solves the issue.

matplotlib 3.7 supports python 3.8, see "list of dependency versions" at [https://matplotlib.org/devdocs/devel/min_dep_policy.html](url)

Upgraded matplotlib to 3.7.4 with pip install --upgrade matplotlib

**Comment by Frank-Duffy** [comment]
After upgrading the package I restarted the new_fret_dev kernel to reflect the upgraded package changes. I re-ran the plot selected fret pairs cell and this time it passed line 3 ax=plt.subplot(111,projection='polar') and failed on the next one since I did not regenerate the data. Error output below:

Upgrading matplotlib to 3.7.4 solved the issue.

---------------------------------------------------------------------------
NameError                                 Traceback (most recent call last)
Cell In[5], line 4
      1 #plot the selected FRET pairs on a radial plot
      3 ax=plt.subplot(111,projection='polar')
----> 4 firstres=single_residue_info['Residue Index'].iloc[0]
      5 lastres=single_residue_info['Residue Index'].iloc[-1]
      6 pairlist=np.array([[i[0],i[1]] for i in Final_Network['Residue Pair']]).astype(float).T

NameError: name 'single_residue_info' is not defined
